app/controllers/concerns/hyku_addons/doi_controller_behavior.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+# TODO: Why is this required, or we see NotFoundError class not found
+module Hyrax
+  module DOI
+    class Error < ::StandardError; end
+    class NotFoundError < Hyrax::DOI::Error; end
+  end
+end
+
+module HykuAddons
+  module DOIControllerBehavior
+    extend ActiveSupport::Concern
+
+    included do
+      def autofill
+        respond_to do |format|
+          format.js { render json: json_response, status: :ok }
+
+          # NOTE: This is temporary, just so we have a URL to debug
+          format.html { render json: json_response, status: :ok }
+        end
+
+      rescue ::Hyrax::DOI::NotFoundError => e
+        respond_to do |format|
+          format.js { render plain: e.message, status: :internal_server_error }
+        end
+      end
+
+      protected
+
+        def json_response
+          {
+            data: formatted_work,
+            curation_concern: curation_concern
+          }.to_json
+        end
+
+        def formatted_work
+          # Download the requested DOI response
+          input = reader_class.new(input: doi)
+
+          # Use that response to build the object as we would expect it
+          meta = reader_class.new(input: input.string)
```

Comment:
@cjcolvar There might be a better way of doing this, if so, could you give me your thoughts?

The DOI request returns a partially formed reader object, with `string` being assigned the result of the response. However the readers all assume that `meta` is available and so the above gets around that by rebuilding the reader with the XML string.